### PR TITLE
feat: adopt @use syntax to support new Sass module system

### PR DESCRIPTION
**Pull request by lucpotage**
This definitely requires additional work but I hope you can find some time to test this PR and improve it. Thank you in advance.

I didn't bumped the npm version.

---
This change is [Reviewable](https://reviewable.io/reviews/shakacode/sass-resources-loader/111)

### CODE REVIEW

**Comment by justin808** [comment]
@lucpotage Any chance that you can take a look at my comments?

**Comment by lucpotage** [comment]
@justin808 I checked how the webpack sass-loader works and it seems they [resolve the implementation](https://github.com/webpack-contrib/sass-loader/commit/ff90dd6) based on the chosen dependency. What about doing the same?

The sass docs says:

> Sass will gradually phase it out over the next few years, and eventually remove it from the language entirely.

What is not clear to me is whether we should force the use of @use in case you have the newest `sass` dependency whereas `@import` is clearly not yet deprecated.

Do you think adding a loader option `rule` would make sense? With `@import` or `@use` possible values and `@use` by default to follow the recommendation.

```js
{
  loader: 'sass-resources-loader',
  options: {
    rule: '@use', // or @import
  },
}
```

Getting feedback from @nex3 would help.

**Comment by justin808** [comment]
@lucpotage I like your last idea. Can you take a try at implementing it?

**Comment by justin808** [comment]
@lucpotage did you see my previous comment?

**Comment by lucpotage** [comment]
Yes and I would like to try implementing it but I definitely don't have the time right now. 😞

File: README.md [hunk 1]
```diff
@@ -67,7 +67,11 @@ module: {
             resources: './path/to/resources.scss',
 
             // Or array of paths
-            resources: ['./path/to/vars.scss', './path/to/mixins.scss']
+            resources: [
+              './path/to/vars.scss',
+              './path/to/mixins.scss',
+              './path/to.functions.scss'
```
Comment (FloEdelmann) [inline README.md:73]:
```suggestion
              './path/to/functions.scss'
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -24,6 +27,7 @@
 For other releases see [GitHub Releases section](https://github.com/shakacode/sass-resources-loader/releases).
 
 [Unreleased]: https://github.com/shakacode/sass-resources-loader/compare/v2.0.0...master
+[2.0.0]: https://github.com/shakacode/sass-resources-loader/compare/v2.0.0...v3.0.0
```
Comment (FloEdelmann) [inline CHANGELOG.md:30]:
```suggestion
[Unreleased]: https://github.com/shakacode/sass-resources-loader/compare/v3.0.0...master
[3.0.0]: https://github.com/shakacode/sass-resources-loader/compare/v2.0.0...v3.0.0
```

File: package.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "sass-resources-loader",
   "version": "2.0.1",
```
Comment (FloEdelmann) [inline package.json:3]:
```suggestion
  "version": "3.0.0",
```

**Comment by justin808** [comment]
We've got some merge conflicts. And some CI issues.

Let me know when this is ready. 

@lucpotage  and @FloEdelmann, thank you for your efforts! Feel free to email me justin@shakacode.com. I've got a slack channel for this as well. You can find the link for Slack here: https://www.shakacode.com/open-source-projects.

**Comment by justin808** [comment]
See #117.

**Pull request closed without merging**